api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -750,22 +750,52 @@ message ScopedRds {
       [(validate.rules).message = {required: true}];
 }
 
+// [#next-free-field: 6]
 message HttpFilter {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.config.filter.network.http_connection_manager.v2.HttpFilter";
 
+  // [#not-implemented-hide:] Configuration source specifier for the late-bound
+  // filter configuration. The HTTP Listener is warmed until all the initial
+  // filter configurations are received, unless the flag to apply the default
+  // configuration is set. Subsequent filter updates are atomic, and apply to
+  // new streams while the active streams continue using the older filter
+  // configurations.
+  // If the initial delivery of the filter configuration fails, due to a
+  // timeout for example, the optional default configuration is applied.
+  // Without a default configuration, the filter is disabled, and the HTTP
+  // listener responds with 500 immediately. After the failure, the listener
+  // continues subscribing to the subsequent filter configurations.
+  message HttpFilterConfigSource {
+    config.core.v3.ConfigSource config_source = 1;
+
+    // Optional default configuration to use as the initial configuration if
+    // there is a failure to receive the initial filter configuration or if
+    // `apply_default_config_without_warming` flag is set.
+    google.protobuf.Any default_config = 2;
+
+    // Use the default config as the initial configuration without warming and
+    // waiting for the first xDS response. Requires the default configuration
+    // to be supplied.
+    bool apply_default_config_without_warming = 3;
+  }
+
   reserved 3, 2;
 
   reserved "config";
 
-  // The name of the filter to instantiate. The name must match a
-  // :ref:`supported filter <config_http_filters>`.
+  // The name of the filter configuration. The name is used as a fallback to
+  // select an extension if the type of the configuration proto is not
+  // sufficient. It also serves as a resource name in FilterConfigDS.
```

Comment:
Do we still have situations where the type of the configuration proto is not sufficient in v3?

Reply:
I don't believe we have in base envoy. Third party filters would need to be migrated but that's not hard to create a stub proto.